### ISSUE TITLE
syntax/make.vim: highlight grouped targets

### ISSUE THREAD

**Issue by matteocoder** [vim/vim#10694]
GNU Make 4.3 introduced 'Grouped targets', which allow to specify that two or more targets are produced by a single recipe simultaneously. Currently, the make syntax file does not support this feature, so I modified the regex to correctly detect it.
The following patch works for me on Vim 9.0.44:
```diff
--- "/c/Program Files/Vim/vim90/syntax/make.vim"	2022-07-02 00:01:18.000000000 +0200
+++ make.vim	2022-07-03 21:32:43.424931600 +0200
@@ -45,11 +45,11 @@
 syn match makeImplicit		"^\.[A-Za-z0-9_./\t -]\+\s*:[^=]"me=e-2
 
 syn region makeTarget transparent matchgroup=makeTarget
-	\ start="^[~A-Za-z0-9_./$()%-][A-Za-z0-9_./\t $()%-]*:\{1,2}[^:=]"rs=e-1
+	\ start="^[~A-Za-z0-9_./$()%-][A-Za-z0-9_./\t $()%-]*\(&\)\?:\{1,2}[^:=]"rs=e-1
 	\ end=";"re=e-1,me=e-1 end="[^\\]$"
 	\ keepend contains=makeIdent,makeSpecTarget,makeNextLine,makeComment,makeDString
 	\ skipnl nextGroup=makeCommands
-syn match makeTarget		"^[~A-Za-z0-9_./$()%*@-][A-Za-z0-9_./\t $()%*@-]*::\=\s*$"
+syn match makeTarget		"^[~A-Za-z0-9_./$()%*@-][A-Za-z0-9_./\t $()%*@-]*\(&\)\?::\=\s*$"
 	\ contains=makeIdent,makeSpecTarget,makeComment
 	\ skipnl nextgroup=makeCommands,makeCommandError
 
```
I tried contacting the maintainer at the address provided in the file,
but so far he has not responded to my mail.

**Comment by jamessan** [comment]
@rohieb

**Comment by rohieb** [comment]
Mh yes sorry, my inbox is an unstructured mess… But the patch looks good to me.

**Comment by rohieb** [comment]
Ah, they also added the `&::` syntax, see https://www.gnu.org/software/make/manual/html_node/Multiple-Targets.html at the bottom. Can you add another `:\?` to your patch?

**Comment by matteocoder** [comment]
@rohieb Do you mean like this?
```diff
--- "/c/Program Files/Vim/vim90/syntax/make.vim"	2022-07-07 00:01:26.000000000 +0200
+++ make.vim	2022-07-14 20:55:08.112473200 +0200
@@ -45,11 +42,11 @@
 syn match makeImplicit		"^\.[A-Za-z0-9_./\t -]\+\s*:[^=]"me=e-2
 
 syn region makeTarget transparent matchgroup=makeTarget
-	\ start="^[~A-Za-z0-9_./$()%-][A-Za-z0-9_./\t $()%-]*:\{1,2}[^:=]"rs=e-1
+	\ start="^[~A-Za-z0-9_./$()%-][A-Za-z0-9_./\t $()%-]*\(&\)\?:\?:\{1,2}[^:=]"rs=e-1
 	\ end=";"re=e-1,me=e-1 end="[^\\]$"
 	\ keepend contains=makeIdent,makeSpecTarget,makeNextLine,makeComment,makeDString
 	\ skipnl nextGroup=makeCommands
-syn match makeTarget		"^[~A-Za-z0-9_./$()%*@-][A-Za-z0-9_./\t $()%*@-]*::\=\s*$"
+syn match makeTarget		"^[~A-Za-z0-9_./$()%*@-][A-Za-z0-9_./\t $()%*@-]*\(&\)\?::\=\s*$"
 	\ contains=makeIdent,makeSpecTarget,makeComment
 	\ skipnl nextgroup=makeCommands,makeCommandError
```

**Comment by rohieb** [comment]
@matteocoder yes, for example. I don't see why you added grouping parentheses around the `&`, but it doesn't hurt either.

**Comment by matteocoder** [comment]
@rohieb Ah, you are right, you can omit them if you want. I tend to be rather paranoid about regexes.

**Comment by brammool** [comment]
> @rohieb Do you mean like this?
> ```diff
> --- "/c/Program Files/Vim/vim90/syntax/make.vim"	2022-07-07 00:01:26.000000000 +0200
> +++ make.vim	2022-07-14 20:55:08.112473200 +0200
> @@ -45,11 +42,11 @@
>  syn match makeImplicit		"^\.[A-Za-z0-9_./\t -]\+\s*:[^=]"me=e-2
>  
>  syn region makeTarget transparent matchgroup=makeTarget
> -	\ start="^[~A-Za-z0-9_./$()%-][A-Za-z0-9_./\t $()%-]*:\{1,2}[^:=]"rs=e-1
> +	\ start="^[~A-Za-z0-9_./$()%-][A-Za-z0-9_./\t $()%-]*\(&\)\?:\?:\{1,2}[^:=]"rs=e-1
>  	\ end=";"re=e-1,me=e-1 end="[^\\]$"
>  	\ keepend contains=makeIdent,makeSpecTarget,makeNextLine,makeComment,makeDString
>  	\ skipnl nextGroup=makeCommands
> -syn match makeTarget		***@***.***[A-Za-z0-9_./\t ***@***.****::\=\s*$"
> +syn match makeTarget		***@***.***[A-Za-z0-9_./\t ***@***.****\(&\)\?::\=\s*$"
>  	\ contains=makeIdent,makeSpecTarget,makeComment
>  	\ skipnl nextgroup=makeCommands,makeCommandError
> ```

Roland - can you please send me the updated file?

-- 
The primary purpose of the DATA statement is to give names to constants;
instead of referring to pi as 3.141592653589793 at every appearance, the
variable PI can be given that value with a DATA statement and used instead
of the longer form of the constant.  This also simplifies modifying the
program, should the value of pi change.
	-- FORTRAN manual for Xerox Computers

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by rohieb** [comment]
This is included in #10774.
